templates/etc/apt/sources.list.j2

```diff
@@ -27,7 +27,7 @@
 {%     endfor %}
 {%   else %}
 {%     for entry_uri in (entry.uri|d([]) + entry.uris|d([])) %}
-{%       if entry_uri in apt__tpl_seen_sources %}
+{%       if (entry_uri in apt__tpl_seen_sources and apt__sources_filter_duplicates|bool) %}
```

Comment:
Have not checked the template in depth yet but maybe it would be better to generate a list of dicts with comments and lines and then use this to create the file and do the duplicate line checking?